lib/oli_web/controllers/page_delivery_controller.ex

```diff
@@ -389,37 +390,42 @@ defmodule OliWeb.PageDeliveryController do
     user = conn.assigns.current_user
 
     resource_attempt = Enum.at(context.resource_attempts, 0)
-    {:ok, resource_attempt_state} = Jason.encode(resource_attempt.state)
-
-    {:ok, activity_guid_mapping} =
-      context.activities
-      |> Jason.encode()
 
     {:ok, {previous, next, _}, _} =
       Oli.Delivery.PreviousNextIndex.retrieve(section, context.page.resource_id)
 
+    previous_url = url_from_desc(conn, section_slug, previous)
+    next_url = url_from_desc(conn, section_slug, next)
+
     render(conn, "advanced_delivery.html", %{
-      review_mode: context.review_mode,
-      graded: context.page.graded,
+      app_params: %{
```

Comment:
By having all the js-app params in one map like this, it becomes more clear what is using which assign.